l1-contracts/contracts/state-transition/StateTransitionManager.sol

```diff
@@ -347,29 +367,28 @@ contract StateTransitionManager is IStateTransitionManager, ReentrancyGuard, Own
         {
             // check input
             bytes32 cutHashInput = keccak256(_diamondCut);
-            require(cutHashInput == initialCutHash, "STM: initial cutHash mismatch");
-        }
-        bytes memory mandatoryInitData;
-        {
-            // solhint-disable-next-line func-named-parameters
-            mandatoryInitData = bytes.concat(
-                bytes32(_chainId),
-                bytes32(uint256(uint160(BRIDGE_HUB))),
-                bytes32(uint256(uint160(address(this)))),
-                bytes32(protocolVersion),
-                bytes32(uint256(uint160(_admin))),
-                bytes32(uint256(uint160(validatorTimelock))),
-                _baseTokenAssetId,
-                bytes32(uint256(uint160(_sharedBridge))),
-                storedBatchZero
-            );
+            if (cutHashInput != initialCutHash) {
+                revert HashMismatch(initialCutHash, cutHashInput);
+            }
         }
 
         // construct init data
         bytes memory initData;
         /// all together 4+9*32=292 bytes for the selector + mandatory data
         // solhint-disable-next-line func-named-parameters
-        initData = bytes.concat(IDiamondInit.initialize.selector, mandatoryInitData, diamondCut.initCalldata);
+        initData = bytes.concat(
```

Comment:
I think you ( or somebody else) did this separation on purpose. Both are fine, just noting

Reply:
It was probably done because of the stack too deep error. In any case, I took the version that was in dev / audited